agent/src/main/java/com/teamscale/jacoco/agent/commit_resolution/git_properties/GitPropertiesLocatorUtils.java

```diff
@@ -42,24 +43,47 @@ public static String getRevisionFromGitProperties(File jarFile) throws IOExcepti
 				entryWithProperties.getFirst(), jarFile);
 	}
 
+	/**
+	 * Tries to extract a file system path to a search root for the git.properties search. A search root is either a
+	 * file system folder or a Jar file. If no such path can be extracted, returns null.
+	 *
+	 * @throws URISyntaxException under certain circumstances if parsing the URL fails. This should be treated the same
+	 *                            as a null search result but the exception is preserved so it can be logged.
+	 */
+	public static File extractGitPropertiesSearchRoot(URL jarOrClassFolderUrl) throws URISyntaxException {
+		String protocol = jarOrClassFolderUrl.getProtocol().toLowerCase();
+		if (protocol.equals("file") && org.conqat.lib.commons.string.StringUtils.endsWithOneOf(
+				jarOrClassFolderUrl.getPath().toLowerCase(), ".jar", ".war", ".ear", ".aar")) {
+			return new File(jarOrClassFolderUrl.toURI());
+		} else if (protocol.equals("jar")) {
+			// used e.g. by Spring Boot. Example: jar:file:/home/k/demo.jar!/BOOT-INF/classes!/
+			String urlString = jarOrClassFolderUrl.toString();
+			String path = urlString.substring(9, urlString.indexOf("!/"));
```

Comment:
Magic number. Possibly better `urlString.indexOf(":")` (in case my eye counting is correct ;-) ?

Reply:
Yeah, I was lazy. I solved it with a regex, which I think is the best solution. That documents the full format of the URL we're parsing

Reply:
Plus it works on Windows where I think the path doesn't start with a slash but with "C:" etc.